File: TDDMicroExercises/PHP/UnicodeFileToHtmlTextConverter/UnicodeFileToHtmlTextConverter.php
<?php

namespace TDDMicroExercises\PHP\UnicodeFileToHtmlTextConverter;

class UnicodeFileToHtmlTextConverter 
{
    private $_fullFilenameWithPath;

    public function __construct($fullFilenameWithPath)
    {
        $this->_fullFilenameWithPath = $fullFilenameWithPath;
    }

    public function convertToHtml()
    {
        $unicodeFileStrem = fopen($this->_fullFilenameWithPath);
        $html = '';

        while ( $line = fgets($unicodeFileStrem)) 
        {
            $html .= htmlentities($line);
            $html .= "<br />";
        }

        return $html;
    }
}

File: TDDMicroExercises.YoursSolutions/CSharp-benwu/UnicodeFileToHtmlTextConverter/UnicodeFileToHtmlTextConverter.cs
using System.IO;
using System.Web;

namespace TDDMicroExercises.UnicodeFileToHtmlTextConverter
{
    public class UnicodeFileToHtmlTextConverter
    {
        protected readonly string FullFilenameWithPath;


        public UnicodeFileToHtmlTextConverter(string fullFilenameWithPath)
        {
            FullFilenameWithPath = fullFilenameWithPath;
        }

        public string ConvertToHtml()
        {
            using (TextReader unicodeFileStream = OpenTextReader())
            {
                string html = string.Empty;

                string line = unicodeFileStream.ReadLine();
                while (line != null)
                {
                    html += HttpUtility.HtmlEncode(line);
                    html += "<br />";
                    line = unicodeFileStream.ReadLine();
                }

                return html;
            }
        }

        protected virtual TextReader OpenTextReader()
        {
            return File.OpenText(FullFilenameWithPath);
        }
    }
}
